components/resource_metering/src/reporter/mod.rs

```diff
@@ -87,8 +88,14 @@ impl Reporter {
     }
 
     fn handle_records(&mut self, records: Arc<RawRecords>) {
-        self.records.append(records);
-        self.records.keep_top_k(self.config.max_resource_groups);
+        let mut records = records.deref().clone();
+        let others = records.keep_top_k(self.config.max_resource_groups);
```

Comment:
Why is records a `Arc<...>` so that we need a clone here? Is it possible to avoid that clone?

Reply:
1. The reason why records is `Arc` is that we may have multiple `Collector`s to share this read-only records.
2. Yes, I'm trying to let `top_k()` to return tow Iterators.